### ISSUE TITLE
fsolve not making progress

### ISSUE THREAD

**Issue by AlexandreSollaci** [rickecon/OGcourse_F16#8]
Hello everyone,

I'm trying to solve for the steady state in problem set 2 and I keep getting an error message that fsolve is not making progress.

"RuntimeWarning: The iteration is not making good progress, as measured by the 
  improvement from the last ten iterations.
  warnings.warn(msg, RuntimeWarning)"

  The code I'm using is

````
def SS_eqns(bvec_guess, *params):
    K = sum(bvec_guess)
    r = alpha*A*(L/K)**(1-alpha)
    w = (1-alpha)*A*(K/L)**(alpha)

    c = np.zeros(3)
    c[0] = w - bvec_guess[0] 
    c[1] = w + (1+r)*bvec_guess[0] - bvec_guess[1]
    c[2] = 0.2*w + (1 + r)*bvec_guess[1]
    
    EulErr_ss = np.zeros(2)
    EulErr_ss[0] = ( beta*(1+r)*c[1]**(-sigma) ) / ( c[0]**(-sigma) ) - 1
    EulErr_ss[1] = ( beta*(1+r)*c[2]**(-sigma) ) / ( c[1]**(-sigma) ) - 1

    return EulErr_ss
    
x = opt.fsolve(SS_eqns, bvec_guess, args=params, xtol = SS_tol)

````

Is anyone having the same issue and knows how to solve it?

**Comment by rickecon** [comment]
I got your code to work by making the following changes.

1) Make sure that bvec_guess is a numpy array: `bvec_guess = np.array([0.1, 0.1])`
2) Make sure that your params tuple is declared before the `opt.fsolve` and unpacked inside your `SS_eqns` function.
3) Also make sure that your specifcation of the `params` tuple that you pass in the `fsolve` command is enclosed in parenthesis. 
4) Lastly, I used the difference specification of the Euler errors rather than the ratio specification that you use. The reason this probably matters is that you do not have any of the "duct tape" that I described in class. That is, the ratio specification of the Euler errors might get you closer to the boundaries of feasibility faster than the difference specification, thereby creating more instances of the fsolve choosing values of `bvec` that are not feasible.
```python
def SS_eqns(bvec_guess, *params):
    beta, sigma, L, A, alpha = params
    ...
    EulErr_ss = np.zeros(2)
    EulErr_ss[0] = beta*(1+r)*c[1]**(-sigma) -  c[0]**(-sigma)
    EulErr_ss[1] = beta*(1+r)*c[2]**(-sigma) -  c[1]**(-sigma)
    return EulErr_ss
...
params = (beta, sigma, L, A, alpha)
x = opt.fsolve(SS_eqns, bvec_guess, args=(params), xtol = SS_tol)
```
Those changes worked on my computer. Let me know if they work for you.
